bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PacketProcessorBase.java

```diff
@@ -119,12 +121,17 @@ protected void sendResponse(int rc, Object response, OpStatsLogger statsLogger)
         }
 
         if (channel.isActive()) {
-            channel.writeAndFlush(response, channel.voidPromise());
+            ChannelPromise promise = channel.newPromise().addListener(future -> {
+                if (!future.isSuccess()) {
+                    logger.debug("Netty channel write exception. ", future.cause());
+                }
+            });
```

Comment:
This will cause unnecessary object creation. If the only purpose is to log exceptions, `channel.voidPromise()` already takes care of that.

Reply:
Thanks for reminder.

Reply:
It has been fixed here: https://github.com/apache/bookkeeper/pull/3733